lib/plugins/aws/package/compile/functions/index.test.js

```diff
@@ -71,6 +71,211 @@ describe('AwsCompileFunctions', () => {
     };
   });
 
+  describe('when function versions are used with layers in the same stack', () => {
+    let firstCfTemplate;
+
+    before(() => {
+      return runServerless({
+        cwd: fixtures.map.functionLayers,
+        cliArgs: ['package'],
+        awsRequestStubMap: {
+          CloudFormation: {
+            describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+          },
+        },
+      }).then(data => {
+        firstCfTemplate = data.cfTemplate;
+      });
+    });
+
+    after(fixtures.cleanup);
+
+    it('should create different version ids for identical lambdas with and without layers', () => {
+      expect(firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref).to.not.equal(
+        firstCfTemplate.Outputs.BarLambdaFunctionQualifiedArn.Value.Ref
+      );
+    });
+
+    it('should create different version ids for identical lambdas with different layer zip content', () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+
+      const functionLayersBasePath = fixtures.map.functionLayers;
+      const originalLayer = path.join(functionLayersBasePath, 'testLayer');
+      const sourceChangeLayer = path.join(
+        functionLayersBasePath,
+        'extra_layers',
+        'testLayerSourceChange'
+      );
+      const backupLayer = path.join(functionLayersBasePath, 'extra_layers', 'testLayerBackup');
```

Comment:
Two questions for this:

1. From what I can see this line https://github.com/serverless/test/blob/0028b7ecddbc815c3ae8f996d7e11d3f7a0dc22b/get-fixtures-engine.js#L106 gives each tested fixture a unique service name, and somehow this is giving the functions unique names too. Therefore, running two separate `runServerless` instances with `fixture: 'functionLayer'` will always produce different version ids. Any way around this?
2. I'm having trouble getting the `fixtures.setup` approach to work consistently. For example, I have a working group of tests using `'fixture: functionLayer`, but if I add even a simple implementation like below it starts affecting my other tests. Any idea what I'm doing wrong?

```
      it('should pass and not affect other tests', () => {
        fixtures
          .setup('functionLayers')
          .then(({ servicePath }) => {
            runServerless({
                cwd: servicePath,
                cliArgs: ['package'],
                awsRequestStubMap: {
                  CloudFormation: {
                    describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
                  },  
                },  
              })  
            }); 
      }); 
```

Reply:
> Any way around this?

Yes. We should setup _fixture_ once, and invoke `runServerless` against same prepared `servicePath` twice. 

> Any idea what I'm doing wrong?

If I see correctly, you do not return result promise, and that makes _test_ case resolved immediately and another one invoked simultaneously.

Since we're on v2 right now. Please switch to async/await syntax, that should help to avoid that issue

Reply:
That solved the second one. For the first, I can get that approach working with fixtures without layers (i.e. the `function` fixture), but when I try use my `functionLayers` fixture the second `runServerless` doesn't seem to generate the layer artifact:

```
Error: ENOENT: no such file or directory, open '/private/var/folders[...]/tmpdirs-serverless/36ba/7738ee/.serverless/testLayer.zip'
```

After the first `runServerless` run the temporary `[..]/36ba/7738ee/.serverless/` directory looks like this:

```
["TestLayerWithCapitals.zip","cloudformation-template-create-stack.json","cloudformation-template-update-stack.json","serverless-state.json","test-functionLayers-k2dh08.zip","testLayer.zip","testLayerWithNoName.zip"]
```

but after the second fails it only has:

```
["cloudformation-template-create-stack.json"]
```

So I'm not sure if there is something wrong with my fixture. The errors above were created using the code below. I also tried to use the async/await approach based on `cognitoUserPool.test.js` so let me know if that's along the right lines.

```
      let servicePath;

      before(async () => {
        const serviceData = await fixtures.setup('functionLayers');
        ({ servicePath } = serviceData);
      }); 
  
  
      it('should pass', async () => {
        await runServerless({
          cwd: servicePath,
          cliArgs: ['package'],
          awsRequestStubMap: {
            CloudFormation: {
              describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
            },  
          },  
        }); 
  
        await runServerless({
          cwd: servicePath,
          cliArgs: ['package'],
          awsRequestStubMap: {
            CloudFormation: {
              describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
            },  
          },  
        }); 
      }); 
```

Reply:
@pwithams that's wierd, both calls should produce same result imo. Can you commit this test setup, so I can reproduce it on my side?

Reply:
I merged in the latest master changes/npm i and it's now working. The other comments look good so I made those changes. Thanks!